compute_tools/src/bin/fast_import.rs

```diff
@@ -52,6 +52,8 @@ struct Args {
     s3_prefix: Option<s3_uri::S3Uri>,
     #[clap(long)]
     source_connection_string: Option<String>,
+    #[clap(long)]
+    restore_connection_string: Option<String>, // will not run postgres if specified, will do pg_restore to this connection string
```

Comment:
I think it's time to refactor things a bit to reflect the fact that there are two operation modes:
1. Restore to local postgres and upload layers to S3.
2. Restore to a running postgres instance.

It's getting hard to read this since it's just one code path where each block is enabled/disabled based on the initial argument set.

This can be done with separate subcommands (see `control_plane/storcon_cli/src/main.rs`):
* Each subcommand has its own set of arguments.
* Each subcommand has it's separate handler. You can share utility functions between the handlers.

Reply:
refactored